src/main/java/net/sf/jabref/model/EntryTypes.java

```diff
@@ -147,6 +149,37 @@ private static void addOrModifyEntryType(EntryType name) {
     }
 
     /**
+     * Determine all CustomTypes which are not overwritten standard types but real custom types for a given BibDatabaseMode
+     *
+     * I.e., a modified "article" type will not be included in the list, but an EntryType like "MyCustomType" will be included.
+     *
+     * @param mode the BibDatabaseMode to be checked
+     * @return  the list of all found custom types
+     */
+    public static List<EntryType> getAllCustomTypes(BibDatabaseMode mode) {
+        Collection<EntryType> allTypes = getAllValues(mode);
+        if(mode == BibDatabaseMode.BIBTEX) {
+            return allTypes.stream().filter(entryType -> !BibtexEntryTypes.getType(entryType.getName()).isPresent())
```

Comment:
Sorry, but I cannot come up with an intelligent solution for such a helper method which is not basically the same as I already wrote here :wink: